### PR TITLE
[Reviews Needed] RPCS3 QT

### PR DESCRIPTION
**Pull request by flash-fire**
Making this PR so we can collect work on converting to Qt in one place.

Please report issues with the build [here ](https://github.com/RPCS3/rpcs3/issues/2711) so that things aren't duplicated.

### CODE REVIEW

**Comment by flash-fire** [comment]
Currently, to build the qt portion with visual studio, here's what you do.

Assumptions, you have the Qt visual studio plugin installed as well as Qt.  Otherwise, you can download it from Qt's website. We haven't finalized Qt's version for RPCS3 (or I haven't been told one yet), but 5.8 works for me.

1) Open rpcs3.sln in visual studio
2) Assuming you have Qt visual studio plugin installed, go to QT5->Open Project File (.pro)
3) Navigate to rpcs3/rpcs3qt and select the qt .pro file. 
4) Right click the now uploaded project rpcs3qt, and click "Set as Startup Project"

**Comment by hcorion** [comment]
@flash-fire Perhaps you could add a README.md in the rpcs3/rpcs3-qt folder with the steps you just explained above.

**Comment by flash-fire** [comment]
@hcorion That sounds good. I'm setting up RPCS3 on my Linux VM later today.  And, I'll add the readme after I figure out Linux build for this.  I don't want to break Linux build when working on windows primarily.

**Comment by hcorion** [comment]
I just tested, and the linux instructions seem to be as easy as:
1. cd to `rpcs3/rpcs3-qt`
2. run `qmake ./rpcs3qt.pro`
3. run `make`
4. run `./rpcs3-qt` to run the gui.

**Comment by flash-fire** [comment]
Thanks for test hcorion.  I'll add that soon enough.

For those curious, I have a branch where wxwidgets is completely removed, and rpcs3 compiles (without the rpcs3 project but with the qt project).  It's located here.  For now, it's probably a novelty.  But, once we're ready, it'll become part of the PR.

[No Wxwidgets branch](https://github.com/flash-fire/rpcs3/tree/RPCS3-NOWX)

**Comment by xsacha** [comment]
Cool, glad that it served a purpose and sorry I never got around to finishing that frontend.

**Comment by flash-fire** [comment]
Yeah. The pro file you made was particularly useful.  Using the plugin's convert to qt project broke some things when using it on the main project.  So, I remade the rpcs3 project from scratch using that project import..

**Comment by twdarkeh** [comment]
Spoke to megamouse about it, but couldn't find you on Discord, so wanted to ask here:

What's the point of having the game list refresh every 10 seconds? Seems like a waste of effort.

**Comment by xsacha** [comment]
PRO files are really simple but I use CMake for everything nowadays ;) Much easier to integrate.

**Comment by flash-fire** [comment]
CMake has actually been a real pain for this Qt transition.  I and a couple other people haven't succeeded in fixing it yet.

(I figured the lambdas were alright.  I just normally try to avoid nesting lambdas)

**Comment by Nekotekina** [comment]
Don't delete /Gui files yet.
`rpcs3/rpcs3qt` is pretty OK location.
Also I would recommend to change file name style:
`rpcs3/rpcs3qt/pad_settings_dialog.cpp`

File: rpcs3/BasicKeyboardHandler.cpp [hunk 1]
```diff
@@ -18,94 +17,109 @@ void BasicKeyboardHandler::Init(const u32 max_connect)
...
 	// Meta Keys
-	m_keyboards[0].m_buttons.emplace_back(WXK_CONTROL, CELL_KB_MKEY_L_CTRL);
-	m_keyboards[0].m_buttons.emplace_back(WXK_SHIFT, CELL_KB_MKEY_L_SHIFT);
-	m_keyboards[0].m_buttons.emplace_back(WXK_ALT, CELL_KB_MKEY_L_ALT);
-	m_keyboards[0].m_buttons.emplace_back(WXK_WINDOWS_LEFT, CELL_KB_MKEY_L_WIN);
-	m_keyboards[0].m_buttons.emplace_back(WXK_COMMAND, CELL_KB_MKEY_L_WIN);
+	m_keyboards[0].m_buttons.emplace_back(Qt::Key_Control, CELL_KB_MKEY_L_CTRL);
+	m_keyboards[0].m_buttons.emplace_back(Qt::Key_Shift, CELL_KB_MKEY_L_SHIFT);
+	m_keyboards[0].m_buttons.emplace_back(Qt::Key_Alt, CELL_KB_MKEY_L_ALT);
+	////m_keyboards[0].m_buttons.emplace_back(WXK_WINDOWS_LEFT, CELL_KB_MKEY_L_WIN); //TODO?
```
Comment (hcorion) [inline rpcs3/BasicKeyboardHandler.cpp]: Just checking these out, I believe the Qt keys you want are:

`Qt::Key_Super_L` to replace both `WXK_WINDOWS_LEFT` and `WXK_COMMAND`

`Qt::Key_Super_R` to replace `WXK_WINDOWS_RIGHT`

Comment (flash-fire) [inline rpcs3/BasicKeyboardHandler.cpp]: Thanks, I'll commit that now..  If you see anything else (I'm sure there's plenty that'll be revised on this massive PR!), let me know.

**Comment by kamer1337** [comment]
Log Closes after PRessing OK in Settings and Views->Show Log/TTV is checked

**Comment by flash-fire** [comment]
@kamer1337 

I know what causes this.  Should be fixed next time I make a build!

**Comment by flash-fire** [comment]
Build updated ;)

**Comment by raveskirza** [comment]
With new build getting the below error when the AutoSave dialog is displayed 

image

**Comment by flash-fire** [comment]
Let me know if you still get that crash in build 6. It shouldv'e been fixed in the last build.

(I'm going to start version controlling.  Oh. How far I have progressed)

**Comment by kamer1337** [comment]
gui config is saving in wrong folder. it creates bin/GuiConfigs outside, instead in rpcs3 folder

**Comment by flash-fire** [comment]
Latest build (8) should fix direct X crash, GuiConfig folder location, and use a different config name for keyboard settings to prevent clash with wxwidgets.

Hopefully, we can finally get some stablization in terms of how many builds I have to compile.

**Comment by kamer1337** [comment]
when open settings "W LDR: SELF: No NPDRM control info found!" is printed to log.
every log is printed to console ( in wx it wasnt like that )

**Comment by AniLeo** [comment]
Yes, it is the same, you just have different logging levels defaulted

**Comment by Gamedev1909** [comment]
Had no issues i booted all my games messed with every settings QT works really well

**Comment by Nekotekina** [comment]
I don't like changes in System.cpp, System.h but it can't be helper right now.

**Comment by Megamouse** [comment]
I did not like the way it handled things in wx.
If you prefer the old way we can change it back. Or do you have another suggestion

**Comment by bjakja** [comment]
The Legend of Heroes Trails of Cold Steel still freezes at start. Another two games I tested - for me works worse then on wx build. Sound is terrible. Thumb of log scrollbar vanish like in previous version.

**Comment by bjakja** [comment]
On wx build works for me on Sandy Bridge without this option.

**Comment by BrokenFantasy** [comment]
Trails of Cold Steel II randomly freezes during its startup screens. The few times I managed to get far enough to load my save, the framerate dropped into the single digits... Any advice please?

**Comment by flash-fire** [comment]
Can you guys make a separate issue? This PR is closed and more people will see it then. I personally won't be able to investigate this at all for at least a few days..

**Comment by bjakja** [comment]
Using new QT build i got assert everytime I use Vulcan.
"F {rsx::thread} : class std::runtime_error thrown: Assertion failed! Result is C4650720h
(in file c:\rpcs3\rpcs3\emu\rsx\vk\VKProgramBuffer.h:146)"

**Comment by hcorion** [comment]
@bjakja Please don't hijack issues. Please go to the forums or discord and ask there.

**Pull request merged**